From: Jiri Slaby <jslaby@suse.cz>
Date: Thu, 17 Oct 2019 20:47:23 +0200
Subject: Revert "drm/i915/bios: make child device order the priority order"
Patch-mainline: no, under discussion
References: bnc#1152971 fdo#112028

This reverts commit 36a0f92020dc8794d3aa69b7fb4c5d2bf99b0099.

We will have a different solution upstream, I assume.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 drivers/gpu/drm/i915/display/intel_bios.c |   28 +++++++++++++++++-----------
 1 file changed, 17 insertions(+), 11 deletions(-)

--- a/drivers/gpu/drm/i915/display/intel_bios.c
+++ b/drivers/gpu/drm/i915/display/intel_bios.c
@@ -1258,7 +1258,8 @@ static enum port get_port_by_ddc_pin(str
 static void sanitize_ddc_pin(struct drm_i915_private *dev_priv,
 			     enum port port)
 {
-	struct ddi_vbt_port_info *info = &dev_priv->vbt.ddi_port_info[port];
+	const struct ddi_vbt_port_info *info =
+		&dev_priv->vbt.ddi_port_info[port];
 	enum port p;
 
 	if (!info->alternate_ddc_pin)
@@ -1266,10 +1267,12 @@ static void sanitize_ddc_pin(struct drm_
 
 	p = get_port_by_ddc_pin(dev_priv, info->alternate_ddc_pin);
 	if (p != PORT_NONE) {
+		struct ddi_vbt_port_info *i = &dev_priv->vbt.ddi_port_info[p];
+
 		DRM_DEBUG_KMS("port %c trying to use the same DDC pin (0x%x) as port %c, "
 			      "disabling port %c DVI/HDMI support\n",
-			      port_name(port), info->alternate_ddc_pin,
-			      port_name(p), port_name(port));
+			      port_name(p), i->alternate_ddc_pin,
+			      port_name(port), port_name(p));
 
 		/*
 		 * If we have multiple ports supposedly sharing the
@@ -1280,9 +1283,9 @@ static void sanitize_ddc_pin(struct drm_
 		 * Give child device order the priority, first come first
 		 * served.
 		 */
-		info->supports_dvi = false;
-		info->supports_hdmi = false;
-		info->alternate_ddc_pin = 0;
+		i->supports_dvi = false;
+		i->supports_hdmi = false;
+		i->alternate_ddc_pin = 0;
 	}
 }
 
@@ -1304,7 +1307,8 @@ static enum port get_port_by_aux_ch(stru
 static void sanitize_aux_ch(struct drm_i915_private *dev_priv,
 			    enum port port)
 {
-	struct ddi_vbt_port_info *info = &dev_priv->vbt.ddi_port_info[port];
+	const struct ddi_vbt_port_info *info =
+		&dev_priv->vbt.ddi_port_info[port];
 	enum port p;
 
 	if (!info->alternate_aux_channel)
@@ -1312,10 +1316,12 @@ static void sanitize_aux_ch(struct drm_i
 
 	p = get_port_by_aux_ch(dev_priv, info->alternate_aux_channel);
 	if (p != PORT_NONE) {
+		struct ddi_vbt_port_info *i = &dev_priv->vbt.ddi_port_info[p];
+
 		DRM_DEBUG_KMS("port %c trying to use the same AUX CH (0x%x) as port %c, "
 			      "disabling port %c DP support\n",
-			      port_name(port), info->alternate_aux_channel,
-			      port_name(p), port_name(port));
+			      port_name(p), i->alternate_aux_channel,
+			      port_name(port), port_name(p));
 
 		/*
 		 * If we have multiple ports supposedlt sharing the
@@ -1326,8 +1332,8 @@ static void sanitize_aux_ch(struct drm_i
 		 * Give child device order the priority, first come first
 		 * served.
 		 */
-		info->supports_dp = false;
-		info->alternate_aux_channel = 0;
+		i->supports_dp = false;
+		i->alternate_aux_channel = 0;
 	}
 }
 
